### PR TITLE
fix: modify fns to adapt to new safe-stable-stringify package update

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- `clone` now returns `null` when the `quickStringify` function returns `undefined`
- `fingerprint` now returns empty string when the `safeStringify` function returns `undefined`

## Context
- Fixes: #18558 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/clone.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { quickStringify } from './stringify';
 
 export function clone<T>(input: T | null = null): T {
-  return JSON.parse(quickStringify(input));
+  const sortedinput = quickStringify(input);
+  return sortedinput ? JSON.parse(sortedinput) : {};
```
Comment (viceice) [inline lib/util/clone.ts:5]:
```suggestion
  return sortedinput ? JSON.parse(sortedinput) : null;
```

we return null if input is null

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — DISMISSED** [review]
don't we need new tests?

File: lib/util/clone.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { quickStringify } from './stringify';
 
 export function clone<T>(input: T | null = null): T {
-  return JSON.parse(quickStringify(input));
+  const sortedinput = quickStringify(input);
```
Comment (viceice) [inline lib/util/clone.ts:4]: is this really sorted? i thought it's only stringified  JSON 

we should add a test that this doesn't change key order.

File: lib/util/fingerprint.spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  it('returns empty string', () => {
+    const res = fingerprint(undefined);
+    expect(res).toBeEmptyString();
+  });
+
+  it('maintains deterministic order', () => {
+    const res = fingerprint(obj);
+    const res2 = fingerprint(obj2);
+    expect(Object.keys(res)).toEqual(Object.keys(res2));
```
Comment (viceice) [inline lib/util/fingerprint.spec.ts:24]:
```suggestion
    expect(res).toEqual(res2);
```
i think you like to compare the strings, not the keys of the string object 🤔

File: lib/util/clone.ts (hunk 2 above)
Comment (RahulGautamSingh) [reply lib/util/clone.ts:4]: "sorted" is misleading actually my bad. It preseves the insertion order only.Test on the way👍

Comment (rarkins) [reply lib/util/clone.ts:4]: rename to `stringifiedInput`?

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**